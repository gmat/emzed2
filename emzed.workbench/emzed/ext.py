import pkg_resources

for ep in pkg_resources.iter_entry_points("emzed_plugin", name="package"):
    print ep
    package = ep.load()
    exec "%s=package" % package.__name__
    del package
    del ep
del pkg_resources

