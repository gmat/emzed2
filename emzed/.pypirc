[distutils]
index-servers = 
  local

[local]
repository: http://127.0.0.1:8080
username:uschmitt
password:pillepalle

