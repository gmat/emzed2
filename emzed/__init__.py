# encoding:latin-1

from version import version as __version__


