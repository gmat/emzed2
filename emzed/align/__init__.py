from rt_align import rtAlign
from mz_align import mzAlign
