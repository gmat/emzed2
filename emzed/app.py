
import pkg_resources
reload(pkg_resources)  # migh be loaded already and is not up to date

for ep in pkg_resources.iter_entry_points("emzed_package", name="main"):
    runner = ep.load()
    print "$$$ loaded", runner
    exec "%s=runner" % ep.module_name.split(".")[0]
    del ep
    del runner

del pkg_resources

