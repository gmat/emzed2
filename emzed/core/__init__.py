from data.Table import Table
