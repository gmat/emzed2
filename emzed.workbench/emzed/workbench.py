def run():
    print 42

def main():
    import os
    print os.system("pip install -i http://127.0.0.1:3142/root/dev/+simple emzed.core")
    import emzed.core
    print emzed.core



