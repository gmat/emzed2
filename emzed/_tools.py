
def is_started_from_cmdline():

    import inspect
    info = inspect.stack()[-1][1]
    result = False
    if info != "<string>":
        if info.endswith("emzed.workbench"):
            result = True
    return result


def gui_running():
    from PyQt4.QtGui import QApplication
    return QApplication.instance() is not None


def is_first_start():
    from emzed.core.config import _UserConfig
    import os
    return not os.path.exists(_UserConfig.config_file_path())
