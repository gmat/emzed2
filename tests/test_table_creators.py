def test_pandas(regtest):
    import pandas
    from emzed.core.data_types import Table
    data = dict(a=[1, 2, 3], b=[1.0, 2.0, None], c=["a", "b", "c"], d=[1.0, 2.0, 3.0],
                e=[None, None, None])
    df = pandas.DataFrame(data, columns=sorted(data.keys()))
    t = Table.from_pandas(df, formats={"b": "%.2f", float: "%.1f", object: "%s", None: "%s"})

    t.print_()
    t.print_(out=regtest)


def test_np_array():
    import numpy as np
    import emzed
    a = np.array([[1, 2, "a"], [1.0, 3, "b"], [2.0, None, np.nan]])
    t = emzed.core.data_types.Table.from_numpy_array(a, ["a", "b", "c"], [float, int, str],
                                                     ["%.2f", "%d", "%s"])

    assert t.rows == [[1.0, 2, "a"], [1.0, 3, "b"], [2.0, None, None]]
