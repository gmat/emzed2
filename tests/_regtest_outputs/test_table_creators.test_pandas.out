a        b        c        d        e       
int      float    str      float    None    
------   ------   ------   ------   ------  
1        1.00     a        1.0      -       
2        2.00     b        2.0      -       
3        -        c        3.0      -       
